src/Environment.jl

```diff
@@ -0,0 +1,29 @@
+
+const _NREFS = Ref(0)
+
+function Init(parts::MPIData;p4est_verbosity_level=P4est_wrapper.SC_LP_DEFAULT)
+  if !MPI.Initialized()
+      MPI.Init()
+  end
+
+  # sc_init(parts.comm, Cint(true), Cint(true), C_NULL, p4est_verbosity_level)
```

Comment:
I think this is the only issue that we have still pending. 

What is the error that we get when we call `sc_init`?

Were we calling `sc_init` before this PR and we were not getting the error?

Reply:
Initializing is not a problem (just added it). What seems to be a problem is finalizing libsc in `Finalize()`. I think it's related to different processes finishing before others. In any case it does not seem to matter if we finalize it or not...